src/Insightly/InsightlyClient.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace CultuurNet\ProjectAanvraag\Insightly;
+
+use CultuurNet\ProjectAanvraag\Insightly\Item\EntityInterface;
+use CultuurNet\ProjectAanvraag\Insightly\Item\Pipeline;
+use CultuurNet\ProjectAanvraag\Insightly\Result\GetPipelinesResult;
+use CultuurNet\ProjectAanvraag\Insightly\Result\GetProjectsResult;
+use Guzzle\Http\ClientInterface;
+use Guzzle\Http\Message\RequestInterface;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Guzzle\Http\Message\Response;
+
+class InsightlyClient implements InsightlyClientInterface
+{
+    /**
+     * @var ClientInterface
+     */
+    protected $guzzleClient;
+
+    /**
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * @var array
+     */
+    private $responseCache = [];
+
+    /**
+     * InsightlyClient constructor.
+     * @param ClientInterface $guzzleClient
+     * @param $apiKey
+     */
+    public function __construct(ClientInterface $guzzleClient, $apiKey)
+    {
+        $this->guzzleClient = $guzzleClient;
+        $this->apiKey = $apiKey;
+    }
+
+    /**
+     * Add filters to the query
+     *
+     * Accepted options:
+     *  - top (int)
+     *  - skip (int)
+     *  - brief (bool)
+     *  - count_total (int)
+     *
+     * @param ParameterBag $query
+     * @param array $options
+     * @return ParameterBag $query
+     */
+    private function addQueryFilters(ParameterBag $query, array $options)
```

Comment:
Construct the parameter bag in this function, instead of constructing a new one when calling this method in getProjects and getPipelines